homeassistant/components/konnected/__init__.py

```diff
@@ -113,352 +61,109 @@
             {
                 vol.Required(CONF_ACCESS_TOKEN): cv.string,
                 vol.Optional(CONF_API_HOST): vol.Url(),
-                vol.Required(CONF_DEVICES): [
-                    {
-                        vol.Required(CONF_ID): cv.matches_regex("[0-9a-f]{12}"),
-                        vol.Optional(CONF_BINARY_SENSORS): vol.All(
-                            cv.ensure_list, [_BINARY_SENSOR_SCHEMA]
-                        ),
-                        vol.Optional(CONF_SENSORS): vol.All(
-                            cv.ensure_list, [_SENSOR_SCHEMA]
-                        ),
-                        vol.Optional(CONF_SWITCHES): vol.All(
-                            cv.ensure_list, [_SWITCH_SCHEMA]
-                        ),
-                        vol.Optional(CONF_HOST): cv.string,
-                        vol.Optional(CONF_PORT): cv.port,
-                        vol.Optional(CONF_BLINK, default=True): cv.boolean,
-                        vol.Optional(CONF_DISCOVERY, default=True): cv.boolean,
-                    }
-                ],
+                vol.Optional(CONF_DEVICES): [DEVICE_SCHEMA_YAML],
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
+YAML_CONFIGS = "yaml_configs"
+PLATFORMS = ["binary_sensor", "sensor", "switch"]
 
-async def async_setup(hass, config):
+
+async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Konnected platform."""
     cfg = config.get(DOMAIN)
     if cfg is None:
         cfg = {}
 
-    access_token = cfg.get(CONF_ACCESS_TOKEN)
+    store = storage.Store(hass, STORAGE_VERSION, STORAGE_KEY)
```

Comment:
In the original implementation the access token was a global value used for all alarm panels.  I don't see a reason it can't be maintained in the cfg entry - will modify.